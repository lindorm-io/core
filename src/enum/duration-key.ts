export enum DurationKey {
  YEARS = "years",
  MONTHS = "months",
  DAYS = "days",
  HOURS = "hours",
  MINUTES = "minutes",
  SECONDS = "seconds",
}
