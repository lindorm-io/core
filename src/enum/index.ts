export * from "./duration-key";
export * from "./node-environment";
