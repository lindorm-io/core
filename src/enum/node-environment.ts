export enum NodeEnvironment {
  PRODUCTION = "production",
  STAGING = "staging",
  DEVELOPMENT = "development",
  TEST = "test",
}
