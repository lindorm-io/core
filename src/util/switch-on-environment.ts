import { NodeEnvironment } from "../enum";
import { TObject } from "../typing";

export interface IEnvironmentObjects {
  production: TObject<any>;
  staging: TObject<any>;
  development: TObject<any>;
  test: TObject<any>;
}

export const switchOnEnvironment = (environment: string, objects: IEnvironmentObjects): TObject<any> => {
  switch (environment) {
    case NodeEnvironment.PRODUCTION:
      return objects.production;

    case NodeEnvironment.STAGING:
      return objects.staging;

    case NodeEnvironment.DEVELOPMENT:
      return objects.development;

    case NodeEnvironment.TEST:
      return objects.test;

    default:
      return {};
  }
};
