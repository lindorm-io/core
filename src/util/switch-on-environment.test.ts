import { IEnvironmentObjects, switchOnEnvironment } from "./switch-on-environment";
import { NodeEnvironment } from "../enum";

describe("switchConfiguration", () => {
  const objects: IEnvironmentObjects = {
    production: { production: true },
    staging: { staging: true },
    development: { development: true },
    test: { test: true },
  };

  test("should return PRODUCTION object", () => {
    expect(switchOnEnvironment(NodeEnvironment.PRODUCTION, objects)).toMatchSnapshot();
  });

  test("should return STAGING object", () => {
    expect(switchOnEnvironment(NodeEnvironment.STAGING, objects)).toMatchSnapshot();
  });

  test("should return DEVELOPMENT object", () => {
    expect(switchOnEnvironment(NodeEnvironment.DEVELOPMENT, objects)).toMatchSnapshot();
  });

  test("should return TEST object", () => {
    expect(switchOnEnvironment(NodeEnvironment.TEST, objects)).toMatchSnapshot();
  });
});
