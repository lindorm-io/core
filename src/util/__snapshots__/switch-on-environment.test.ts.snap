// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switchConfiguration should return DEVELOPMENT object 1`] = `
Object {
  "development": true,
}
`;

exports[`switchConfiguration should return PRODUCTION object 1`] = `
Object {
  "production": true,
}
`;

exports[`switchConfiguration should return STAGING object 1`] = `
Object {
  "staging": true,
}
`;

exports[`switchConfiguration should return TEST object 1`] = `
Object {
  "test": true,
}
`;
